cupy/fft/fft.py

```diff
@@ -373,18 +387,15 @@ def _get_cufft_plan_nd(shape, fft_type, axes=None, order='C', out_size=None):
             raise ValueError(
                 'Invalid number of FFT data points specified.')
 
-    plan = cufft.PlanNd(shape=plan_dimensions,
-                        inembed=inembed,
-                        istride=istride,
-                        idist=idist,
-                        onembed=onembed,
-                        ostride=ostride,
-                        odist=odist,
-                        fft_type=fft_type,
-                        batch=nbatch,
-                        order=order,
-                        last_axis=fft_axes[-1],
-                        last_size=out_size)
+    keys = (plan_dimensions, inembed, istride,
+            idist, onembed, ostride, odist,
+            fft_type, nbatch, order, fft_axes[-1], out_size)
```

Comment:
shouldn't `cupy.cuda.device.get_device_id()` also be in the list of keys here? (and in the single-GPU case for 1D plans above?)

Reply:
I think we need a per-thread, per-device cache. I will do the per-device part.

Reply:
You mean each device will have it's own independent cache? I think that would be best as I don't think it would make sense to combine memory from separate devices when computing the current cache memory usage.

Reply:
Done in 737d0ab.